components/kyma-environment-broker/internal/process/provisioning/monitoring_integration.go

```diff
@@ -0,0 +1,105 @@
+package provisioning
+
+import (
+	"time"
+
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal"
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/broker"
+	kebError "github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/error"
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/monitoring"
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/process"
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/storage"
+	"github.com/kyma-project/control-plane/components/provisioner/pkg/gqlschema"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	MonitoringComponentName = "rma"
+)
+
+type MonitoringIntegrationStep struct {
+	operationManager *process.ProvisionOperationManager
+	client           monitoring.Client
+	monitoringConfig monitoring.Config
+}
+
+func NewMonitoringIntegrationStep(os storage.Operations, client monitoring.Client, monitoringConfig monitoring.Config) *MonitoringIntegrationStep {
+	return &MonitoringIntegrationStep{
+		operationManager: process.NewProvisionOperationManager(os),
+		client:           client,
+		monitoringConfig: monitoringConfig,
+	}
+}
+
+func (s *MonitoringIntegrationStep) Name() string {
+	return "Monitoring_Integration"
+}
+
+func (s *MonitoringIntegrationStep) Run(operation internal.ProvisioningOperation, log logrus.FieldLogger) (internal.ProvisioningOperation, time.Duration, error) {
+	releaseName := operation.InstanceDetails.ShootName
+	isPresent, err := s.client.IsInstalled(releaseName)
```

Comment:
If the last revision of the release is not deployed (e.g. pending, failed) for some reason, this would still return true and the step wouldn't retry installing.
For uninstalling, checking all possible states (deployed, pending, failed, superseded) is fine. But when trying to install I think we only need to consider deployed status (maybe change the existing `IsInstalled()` to `IsPresent()`, and have a separate `IsDeployed()` method?), and (re)try installation with `upgrade` helm option set (as the release might be present).

Reply:
change install process as suggested